File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task4.java

import java.util.HashSet;
import java.util.Set;

public class Task4 {
    public static boolean isIsogram(String str) {
        if (str == null) {
            return true;
        }
        
        String lowerStr = str.toLowerCase();
        Set<Character> seenChars = new HashSet<>();
        
        for (int i = 0; i < lowerStr.length(); i++) {
            char c = lowerStr.charAt(i);
            if (Character.isLetter(c)) {
                if (seenChars.contains(c)) {
                    return false;
                }
                seenChars.add(c);
            }
        }
        
        return true;
    }
    
    public static void main(String[] args) {
        // Test case 1
        System.out.println("Test 1: 'Dermatoglyphics' -> " + isIsogram("Dermatoglyphics"));
        
        // Test case 2
        System.out.println("Test 2: 'aba' -> " + isIsogram("aba"));
        
        // Test case 3
        System.out.println("Test 3: 'moOse' -> " + isIsogram("moOse"));
        
        // Test case 4
        System.out.println("Test 4: '' -> " + isIsogram(""));
        
        // Test case 5
        System.out.println("Test 5: 'isogram' -> " + isIsogram("isogram"));
    }
}

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task4.py

def is_isogram(s):
    if s is None:
        return True
    
    lower_str = s.lower()
    seen_chars = set()
    
    for char in lower_str:
        if char.isalpha():
            if char in seen_chars:
                return False
            seen_chars.add(char)
    
    return True


if __name__ == "__main__":
    # Test case 1
    print(f"Test 1: 'Dermatoglyphics' -> {is_isogram('Dermatoglyphics')}")
    
    # Test case 2
    print(f"Test 2: 'aba' -> {is_isogram('aba')}")
    
    # Test case 3
    print(f"Test 3: 'moOse' -> {is_isogram('moOse')}")
    
    # Test case 4
    print(f"Test 4: '' -> {is_isogram('')}")
    
    # Test case 5
    print(f"Test 5: 'isogram' -> {is_isogram('isogram')}")
